### PR TITLE
EthIo Rev 2

### PR DESCRIPTION
**Pull request by callendorph**
I have not updated the dependencies yet - but they can be found in these PRs:

https://github.com/JITx-Inc/FTDI/pull/6
https://github.com/JITx-Inc/microchip-networking/pull/4
https://github.com/JITx-Inc/stmicro/pull/7
https://github.com/JITx-Inc/diodes/pull/1
https://github.com/JITx-Inc/jsl/pull/194

### CODE REVIEW

**Comment by d-haldane** [comment]
Looks pretty solid overall!

File: src/controller.stanza [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+  net (debug-if.USB, USB)
+
+  ; Connect the Debug Interface to the
+  ;  MCU by requesting the necessary interfaces
+  ;  via pin assignment.
+
+  require acm:uart-fc() from debug-if.ftdi
+  require ctl-uart:uart-fc() from ctl.mcu
+  net (acm, ctl-uart)
```
Comment (bhusang) [inline src/controller.stanza:104]: Are you expecting to connect RX -> RX and TX -> TX here?

Comment (callendorph) [reply src/controller.stanza:104]: Oof - good catch.

Comment (callendorph) [reply src/controller.stanza]: Addressed with the latest

**Comment by callendorph** [comment]
I was able to run the design in a fresh test directory without issue:

```
Loading JITX Shell...
slm.exe: cloning EEPROM at 0.3.2
slm.exe: cloning microchip-networking at 0.5.0
slm.exe: cloning stmicro at 0.4.0
slm.exe: cloning jlc-pcb at 0.3.0
slm.exe: cloning jsl at 0.9.0
slm.exe: cloning FTDI at 0.3.3
slm.exe: cloning debug at 0.3.3
slm.exe: cloning connectors at 0.4.0
slm.exe: cloning TI-vreg at 0.3.1
slm.exe: cloning diodes at 0.2.0
slm.exe: cloning passives at 0.1.0
slm.exe: cloning power-systems at 0.5.0
slm.exe: cloning maybe-utils at 0.1.6
slm.exe: cloning switches at 0.1.0
slm.exe: cloning inductors at 0.3.0
slm.exe: cloning voltage-divider at 0.4.0
  resistance = 16900.0  [934ms] (1) "CQ02WGF1692TCE"
  capacitance = 5.6e-008 to 8.20000000000001e-008  [909ms] (1) "C0402C683J4RAC7867"
  resistance = 1210000.0  [686ms] (1) "CR0402-16W-1214FT"
  resistance = 357000.0  [845ms] (1) "CR0402-16W-3573FT"
  capacitance = 2.7e-006 to 3.3e-006  [1s] (1) "CGA4J1X7R1E335K125AC"
  capacitance = 1.0e-005  [1s] (1) "GRM188Z71A106KA73D"
  capacitance = 4.7e-009  [1s] (1) "0603J0250472JDT"
  capacitance = 1.0e-007  [761ms] (1) "06031C104K4Z2A"
  resistance = 0.0  [503ms] (1) "RC0402JR-130RL"
-> Precision 0.25%
  resistance = 42300.0 to 51700.0  [1s] (19) 46400.0 43200.0 47500.0 48100.0 42700.0 44200.0 51700.0 51000.0 45900.0 47000.0
  resistance = 7200.0 to 8800.0  [1s] (18) 7870.0 7500.0 8060.0 7320.0 8200.0 8560.0 8250.0 8760.0 7960.0 8450.0
    - Querying resistors for R-h=43000.0╬⌐ R-l=7320.0╬⌐
  resistance = 43000.0  [1s] (4) "ERA-2ARC433X" "RN731ETTP4302C50" "RN731ETTP4302C25" "RN731JTTD4302C10"
search-resistors: warning: no distinct key specified for the search.
  resistance = 7320.0  [1s] (4) "ERA-2ARC7321X" "RN731ETTP7321C25" "RN731ETTP7321C50" "RN731JTTD7321C25"
search-resistors: warning: no distinct key specified for the search.
      Solved: mpn1=ERA-2ARC433X, mpn2=ERA-2ARC7321X, v-out=(Toleranced(5.39157229587988 <= 5.49945355191257 <= 5.60825238252809), Toleranced(5.39387465761825 <= 5.49945355191257 <= 5.60585485571853)V), current=0.000751291468840515A
  capacitance = 1.0e-011  [4s] (1) "06033C100KAT2A"
  resistance = 0.0  [786ms] (1) "CQ02WGF0000TCE"
  capacitance = 1.0e-005  [946ms] (1) "GRM188Z71A106KA73D"
  capacitance = 4.7e-009  [455ms] (1) "06035C472KAZ2A"
  capacitance = 2.2e-005  [1s] (1) "885012208019"
-> Precision 0.1%
  resistance = 25650.0 to 31350.0  [725ms] (19) 30000.0 30100.0 30900.0 31200.0 26700.0 27400.0 28700.0 27100.0 27000.0 27700.0
  resistance = 4050.0 to 4950.0  [597ms] (19) 4480.0 4930.0 4120.0 4810.0 4530.0 4640.0 4590.0 4300.0 4270.0 4320.0
    - Querying resistors for R-h=28000.0╬⌐ R-l=4420.0╬⌐
  resistance = 28000.0  [2s] (26) "AT0402BRD0728KL" "ERA-2APB2802X" "ERA-2AEB2802X" "ERA-2ARB2802X" "PTFR0402B28K0P9" "PTFR0402B28K0N9" "RG1005N-2802-B-T5" "RN731ETTP2802B50" "RNCF0402BTC28K0" "RNCF0402BTE28K0"
search-resistors: warning: no distinct key specified for the search.
  resistance = 4420.0  [2s] (38) "AT0402BRD074K42L" "CPF0402B4K42E1" "ERA-2APB4421X" "ERA-2AEB4421X" "ERA-2ARB4421X" "MCS04020D4421BE100" "PTN0402E4421BST1" "PTFR0402B4K42P9" "RG1005P-4421-B-T5" "RG1005N-4421-B-T5"
search-resistors: warning: no distinct key specified for the search.
      Solved: mpn1=AT0402BRD0728KL, mpn2=AT0402BRD074K42L, v-out=(Toleranced(3.20365762194597 <= 3.30067873303167 <= 3.39822407309638), Toleranced(3.20711685599468 <= 3.30067873303167 <= 3.39455608549726)V), current=0.000746759894351058A   
  capacitance = 4.7e-009  [904ms] (1) "06035C472KAZ2A"
  capacitance = 2.2e-005  [380ms] (1) "885012208019"
-> Precision 0.25%
  resistance = 6750.0 to 8250.0  [2s] (19) 7590.0 6800.0 6980.0 7060.0 7230.0 7500.0 6810.0 8160.0 8200.0 8250.0
  resistance = 4050.0 to 4950.0  [446ms] (18) 4640.0 4220.0 4870.0 4700.0 4170.0 4320.0 4930.0 4420.0 4480.0 4370.0
-> Precision 0.1%
  resistance = 6750.0 to 8250.0  [503ms] (20) 7320.0 7770.0 8060.0 7150.0 8200.0 6810.0 7500.0 7060.0 7960.0 8250.0
    - Querying resistors for R-h=8200.0╬⌐ R-l=4930.0╬⌐
  resistance = 8200.0  [1s] (30) "ARG02BTC8201" "CPF0402B8K2E1" "ERA-2ARB822X" "ERA-2APB822X" "ERA-2ARB8201X" "ERA-2VPB8201X" "RG1005N-822-B-T5" "RG1005P-822-B-T10" "RG1005P-822-B-T5" "RT0402BRD078K2L"
search-resistors: warning: no distinct key specified for the search.
  resistance = 4930.0  [850ms] (6) "TNPW04024K93BETD" "RN731JTTD4931B05" "RN731JTTD4931B50" "RN731JTTD4931B10" "TNPW06034K93BEEA" "TNPW06034K93BYEA"
search-resistors: warning: no distinct key specified for the search.
      Solved: mpn1=ARG02BTC8201, mpn2=TNPW04024K93BETD, v-out=(Toleranced(1.16415542487547 <= 1.19847870182556 <= 1.23293962116275), Toleranced(1.16506368682028 <= 1.19847870182556 <= 1.23197654855667)V), current=0.000243099128159342A      
  resistance = 10000.0  [653ms] (1) "AA0402FR-0710KL"
  resistance = 6040.0  [950ms] (1) "CRCW04026K04FKEDC"
  capacitance = 1.0e-007  [755ms] (1) "LMF105B7104KVHF"
  capacitance = 2.2e-005  [950ms] (1) "GRM21BZ71A226ME15L"
  capacitance = 3.6e-011  [1s] (1) "C0402C360J5GACTU"
  resistance = 4700.0  [611ms] (1) "0402 ┬▒1% 4K7"
  resistance = 51.0  [737ms] (1) "ARG02FTC0510"
  capacitance = 4.7e-009  [548ms] (1) "0402YC472KAT2A"
  resistance = 330.0  [723ms] (1) "AF0402FR-07330RL"
  resistance = 10000.0  [2s] (1) "CRCW010010K0JREL"
  capacitance = 4.7e-006  [1s] (1) "GRM188Z71A475KE15D"
  capacitance = 2.2e-006  [652ms] (1) "GRM155Z71A225KE01D"
  capacitance = 1.3e-011  [904ms] (1) "0402YA130JAT2A"
  resistance = 51.0  [445ms] (1) "AC0603FR-0751RL"
  resistance = 12000.0  [584ms] (1) "CR0402-FX-1202GLF"
  capacitance = 1.0e-007  [583ms] (1) "0402B104K100CT"
  capacitance = 3.3e-006  [627ms] (1) "CGA4J3X7R1A335K125AB"
  capacitance = 4.7e-006  [774ms] (1) "GRM188Z71A475KE15D"
  resistance = 2000.0  [661ms] (1) "CQ02WGF2001TCE"
  capacitance = 1.0e-009  [427ms] (1) "0402B102K160CT"
  capacitance = 1.0e-006  [805ms] (1) "04026C105KAT2A"
  resistance = 5100.0  [459ms] (1) "CQ02WGF5101TCE"
  capacitance = 1.0e-007  [548ms] (1) "04025C104KAT4A"
Saving stable design and reference designator table
REPL environment is now inside package ethernet-io/main.
```

**Pull request merged**